pkg/arvo/sys/vane/khan.hoon

```diff
@@ -0,0 +1,222 @@
+      $%  $:  %g                                        ::    to %gall
+              $>(%deal task:gall)                       ::  full transmission
+          ==                                            ::
+          $:  %k                                        ::    to self
+              $>(%fard task)                            ::  internal thread
+      ==  ==                                            ::
+    +$  sign                                            ::    in response $<-
+      $%  $:  %gall                                     ::    from %gall
+              $>(%unto gift:gall)                       ::  update
+          ==                                            ::
+          $:  %khan                                     ::    from self
+              $>(?(%arow %avow) gift)                   ::  thread result
+      ==  ==                                            ::
+    +$  khan-state                                      ::
+      $:  %0                                            ::    state v0
+          hey=duct                                      ::  unix duct
+          tic=@ud                                       ::  tid counter
+      ==                                                ::
+    --                                                  ::
+=>
+|%
+++  get-beak
+  |=  [=bear now=@da]
+  ?@(bear [our bear %da now] bear)
+::
+++  get-dais
+  |=  [=beak =mark rof=roof]
+  ^-  dais:clay
+  ?~  ret=(rof ~ %cb beak /[mark])
+    ~|(mark-unknown+mark !!)
+  ?~  u.ret
+    ~|(mark-invalid+mark !!)
+  ?>  =(%dais p.u.u.ret)
+  !<(dais:clay q.u.u.ret)
+::
+++  get-tube
+  |=  [=beak =mark =out=mark rof=roof]
+  ^-  tube:clay
+  ?~  ret=(rof ~ %cc beak /[mark]/[out-mark])
+    ~|(tube-unknown+[mark out-mark] !!)
+  ?~  u.ret
+    ~|(tube-invalid+[mark out-mark] !!)
+  ?>  =(%tube p.u.u.ret)
+  !<(tube:clay q.u.u.ret)
+::
+++  make-wire
+  |=  [=beak =mark]
+  ^-  wire
+  [%fyrd (en-beam beak mark ~)]
+::
+++  read-wire
+  |=  =wire
+  ^-  (pair beak mark)
+  ~|  khan-read-wire+wire
+  ?>  ?=([%fyrd ^] wire)
+  =/  =beam  (need (de-beam t.wire))
+  ?>(?=([@ ~] s.beam) beam(s i.s.beam))
+::
+++  start-spider
+  |=  [our=@p =vase]
+  ^-  note
+  [%g %deal [our our] %spider %poke %spider-start vase]
+::
+++  watch-spider
+  |=  [our=@p =path]
+  ^-  note
+  [%g %deal [our our] %spider %watch path]
+--
+=|  khan-state
+=*  state  -
+|=  [now=@da eny=@uvJ rof=roof]
+=*  khan-gate  .
+^?
+|%
+::  +call: handle a +task request
+::
+++  call
+  |=  $:  hen=duct
+          dud=(unit goof)
+          wrapped-task=(hobo task)
+      ==
+  ^-  [(list move) _khan-gate]
+  ::
+  =/  =task  ((harden task) wrapped-task)
+  ?^  dud
+    ~|(%khan-call-dud (mean tang.u.dud))
+  ?+    -.task  [~ khan-gate]
+      %born
+    [~ khan-gate(hey hen)]
+  ::
+      %fard
+    =/  tid=@ta
+      %^  cat  3
+        'khan-fyrd--'
+      (scot %uv (sham (mix tic eny)))
+    =.  tic     (mod +(tic) (bex 128))
```

Comment:
This seems like overkill. Not only is it hard to imagine this call `+mod` ever affecting the value of `tic`, but the actual space of collisions that we want to avoid is only within a single event. If you're worried about `tic` growing unbounded, you could just reset it to `0` on `%born`.

(You could alternately persist the entropy in your state on `%born`, and then stretch/iterate it yourself a la `|og`. But that might be overkill for generating ids.)